apps/frontend/src/components/ui/moderation/ModerationTechRevCard.vue

```diff
@@ -0,0 +1,1107 @@
+
+	for (const issue of selectedFile.value.issues) {
+		for (const detail of issue.details) {
+			if (!classMap.has(detail.file_path)) {
+				classMap.set(detail.file_path, { filePath: detail.file_path, flags: [] })
+			}
+			// Cast detail to include status (backend will provide this field)
+			const detailWithStatus = detail as Labrinth.TechReview.Internal.ReportIssueDetail & {
+				status: Labrinth.TechReview.Internal.DelphiReportIssueStatus
+			}
+			classMap.get(detail.file_path)!.flags.push({
+				issueId: issue.id,
+				issueType: issue.issue_type,
+				detail: detailWithStatus,
+			})
+		}
+	}
+
+	for (const classGroup of classMap.values()) {
+		classGroup.flags.sort((a, b) => {
+			const aPreReviewed = isPreReviewed(a.detail.id, a.detail.status)
+			const bPreReviewed = isPreReviewed(b.detail.id, b.detail.status)
+
+			if (aPreReviewed !== bPreReviewed) {
+				return aPreReviewed ? 1 : -1
+			}
+
+			return (severityOrder[b.detail.severity] ?? 0) - (severityOrder[a.detail.severity] ?? 0)
+		})
+	}
+
+	return Array.from(classMap.values()).sort((a, b) => {
+		const aSeverity = getHighestSeverityInClass(a.flags)
+		const bSeverity = getHighestSeverityInClass(b.flags)
+		return (severityOrder[bSeverity] ?? 0) - (severityOrder[aSeverity] ?? 0)
+	})
+})
+
+function getHighestSeverityInClass(
+	flags: ClassGroup['flags'],
+): Labrinth.TechReview.Internal.DelphiSeverity {
+	return flags.reduce(
+		(highest, flag) =>
+			(severityOrder[flag.detail.severity] ?? 0) > (severityOrder[highest] ?? 0)
+				? flag.detail.severity
+				: highest,
+		'low' as Labrinth.TechReview.Internal.DelphiSeverity,
+	)
+}
+
+function toggleClass(filePath: string) {
+	if (expandedClasses.value.has(filePath)) {
+		expandedClasses.value.delete(filePath)
+	} else {
+		expandedClasses.value.add(filePath)
+	}
+}
+
+function handleThreadUpdate() {
+	emit('refetch')
+}
+
+const techReviewContext = computed<TechReviewContext>(() => ({
+	project: props.item.project,
+	project_owner: props.item.project_owner,
+	reports: props.item.reports,
+}))
+
+const threadViewRef = ref<{
+	setReplyContent: (content: string) => void
+	getReplyContent: () => string
+} | null>(null)
+
+const maliciousSummaryModalRef = ref<InstanceType<typeof MaliciousSummaryModal>>()
+
+const unsafeFiles = computed<UnsafeFile[]>(() => {
+	return props.item.reports
+		.filter((report) =>
+			report.issues.some((issue) =>
+				issue.details.some((detail) => {
+					const detailWithStatus = detail as typeof detail & {
+						status: Labrinth.TechReview.Internal.DelphiReportIssueStatus
+					}
+					const decision = getDetailDecision(detailWithStatus.id, detailWithStatus.status)
+					return decision === 'malware'
+				}),
+			),
+		)
+		.map((report) => ({
+			file: report,
+			projectName: props.item.project.name,
+			projectId: props.item.project.id,
+			userId: props.item.project_owner.id,
+		}))
+})
+
+const reviewSummaryPreview = computed(() => {
+	interface FileDecisions {
```

Comment:
Could bring this interface out of computed, unless theres a specific reason not to